unit_tests/test_cover_plugin.py

```diff
@@ -1,25 +1,71 @@
+
+
+def _get_coverage_obj(args, env=None):
+    parser = OptionParser()
+    c = Coverage()
+    c.addOptions(parser, env)
+    options, args = parser.parse_args(['test_can_be_disabled'] + args)
+    c.configure(options, Config())
+    return c
+
 
 class TestCoveragePlugin(object):
 
     def test_cover_packages_option(self):
-        parser = OptionParser()
-        c = Coverage()
-        c.addOptions(parser)
-        options, args = parser.parse_args(['test_can_be_disabled',
-                                           '--cover-package=pkg1,pkg2,pkg3'])
-        c.configure(options, Config())
+        c = _get_coverage_obj(['--cover-package=pkg1,pkg2,pkg3'])
         eq_(['pkg1', 'pkg2', 'pkg3'], c.coverPackages)
 
         env = {'NOSE_COVER_PACKAGE': 'pkg1,pkg2,pkg3'}
-        c = Coverage()
-        parser = OptionParser()
-        c.addOptions(parser, env)
-        options, args = parser.parse_args(['test_can_be_disabled'])
-        c.configure(options, Config())
+        c = _get_coverage_obj([], env)
         eq_(['pkg1', 'pkg2', 'pkg3'], c.coverPackages)
+
+    def test_want_file_inclusive(self):
+        c = _get_coverage_obj(['--cover-inclusive'])
+        assert_true(c.wantFile('pkg1/foo.py'))
+
+        env = {'NOSE_COVER_INCLUSIVE': 'yes'}
+        c = _get_coverage_obj([], env)
+        assert_true(c.wantFile('pkg1/foo.py'))
+
+    def test_want_file_inclusive_exclude(self):
+        c = _get_coverage_obj(['--cover-inclusive',
+                               '--cover-exclude-file=foo.py'])
+        assert_false(c.wantFile('foo.py'))
+        assert_true(c.wantFile('bar.py'))
+
+        env = {'NOSE_COVER_INCLUSIVE': 'yes',
+               'NOSE_COVER_EXCLUDE_FILE': 'foo.py'}
+        c = _get_coverage_obj([], env)
+        assert_false(c.wantFile('foo.py'))
+        assert_true(c.wantFile('bar.py'))
+
+    def test_want_file_inclusive_exclude_globs(self):
+        c = _get_coverage_obj(['--cover-inclusive',
+                               '--cover-exclude-file=foo/*.py'])
+        assert_false(c.wantFile('foo/bar.py'))
+        assert_true(c.wantFile('bar.py'))
+
+        env = {'NOSE_COVER_INCLUSIVE': 'yes',
+               'NOSE_COVER_EXCLUDE_FILE': 'foo/*.py'}
+        c = _get_coverage_obj([], env)
+        assert_false(c.wantFile('foo/bar.py'))
+        assert_true(c.wantFile('bar.py'))
+
+    def test_want_file_inclusive_multiple_args(self):
+        c = _get_coverage_obj(['--cover-inclusive',
+                               '--cover-exclude-file=foo.py',
+                               '--cover-exclude-file=bar.py'])
+        assert_false(c.wantFile('foo.py'))
+        assert_false(c.wantFile('bar.py'))
+        assert_true(c.wantFile('baz.py'))
+
+        env = {'NOSE_COVER_INCLUSIVE': 'yes',
+               'NOSE_COVER_EXCLUDE_FILE': 'bar.py,foo.py'}
+        c = _get_coverage_obj([], env)
+        assert_false(c.wantFile('foo.py'))
+        assert_false(c.wantFile('bar.py'))
+        assert_true(c.wantFile('baz.py'))
```

Comment:
Nope, this should work for multiple files; the option is probably named
poorly. I'll fix that up this afternoon.

On Wed, Jun 17, 2015 at 10:34 AM John Szakmeister <notifications@github.com>
wrote:

> In unit_tests/test_cover_plugin.py
> <https://github.com/nose-devs/nose/pull/924#discussion_r32609335>:
>
> > +        assert_true(c.wantFile('bar.py'))
> > +
> > +    def test_want_file_inclusive_multiple_args(self):
> > +        c = _get_coverage_obj(['--cover-inclusive',
> > +                               '--cover-exclude-file=foo.py',
> > +                               '--cover-exclude-file=bar.py'])
> > +        assert_false(c.wantFile('foo.py'))
> > +        assert_false(c.wantFile('bar.py'))
> > +        assert_true(c.wantFile('baz.py'))
> > +
> > +        env = {'NOSE_COVER_INCLUSIVE': 'yes',
> > +               'NOSE_COVER_EXCLUDE_FILE': 'bar.py,foo.py'}
> > +        c = _get_coverage_obj([], env)
> > +        assert_false(c.wantFile('foo.py'))
> > +        assert_false(c.wantFile('bar.py'))
> > +        assert_true(c.wantFile('baz.py'))
>
> Thank you for even considering to write tests!
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/nose-devs/nose/pull/924/files#r32609335>.
>